src/components/GraphiQL.js

```diff
@@ -475,7 +475,13 @@ export class GraphiQL extends React.Component {
 
   _fetchQuery(query, variables, operationName, cb) {
     const fetcher = this.props.fetcher;
-    const fetch = fetcher({ query, variables, operationName });
+    const parsedVariables =
+      variables && variables.trim() !== '' ? JSON.parse(variables) : null;
```

Comment:
I think it'd be better to wrap `JSON.parse` in a try/catch block to avoid a console error:
```js
let jsonVariables = null;
try {
  jsonVariables = JSON.parse(variables);
} catch (error) {
  jsonVariables = variables || null;
}

const fetch = fetcher({ ... });
```
What do you think?